### ISSUE TITLE
ability to import already created deck by list.[feature request]

### ISSUE THREAD

**Issue by gardian06** [HearthSim/Hearthstone-Deck-Tracker#2406]
where the current method requires construction of the deck after import, and after about 50 or so tries the system has never worked. instead allow the system to view the list of cards in deck, and create deck based on mana-cost cross name, and then add number.

potential issues: in decks containing more then 21 unique cards some requirement to scroll the decklist may be needed by checking the the presence of the scroll bar, and having the user scroll the list or force the user to import the possibly up to 9 additional cards.

**Comment by Epix37** [comment]
Could you rephrase what you mean? Do you mean detecting the cards using image recognition?

**Comment by gardian06** [comment]
I was thinking something along the lines of moderate image recognition (compare only no guessing)  considering there is not that much different between golden and non-golden cards. and the names do appear to standout relatively well (I'm sure there may be exceptions, but they should be just that)

where the program already has the images of the cards as they appear in lists, and does not need to really guess, the only parts the program needs to recognize is mana cost, and then the count, and where mana cost is also part of the card image we are only talking about 1 real test for recognition being the existance of a 2 or not in the count

I was thinking of first checking for the existence of the scroll bar (if no scroll bar then we can . then using the first third of the card in the list (this would include the mana cost and in most cases a majority of the first word if not the complete thing) then do compares to get mana cost of the first card in the list where the majority of decks start at either 0, 1, or 2 we can find this quickly then use knowledge of the pattern that is used to by the game to order the cards in the list (spells, class minions, minions) walking through the list making sure to check mana cost first just in case it changes (if it is different then test on how much of a jump there was typically 1)

explaining further gets complicated for me to follow so I created a flowchart of what I was thinking. it looks a lot cleaner in openOfficeDraw, but was able to get to jpeg


![subroutineflow](https://cloud.githubusercontent.com/assets/15876343/14905467/b793731e-0d65-11e6-97a3-2006b36ebcce.jpg)

**Comment by mraakashshah** [comment]
This seems like an import from Hearthstone screenshot function?

**Comment by gardian06** [comment]
if the entire deck fits into 1 screenshot then yes that analogy could be used, but if the scroll bar exists (more then 21 unique cards) then a single screen shot will not work. either the user will need to scroll the bar, or have to enter the cards manually. 

from a user point of view: if you have the image of a decklist at highest focus, and in the expected possition of a decklist then it might be possible to have this import system import that deck. not sure if you are talking about for the perpose of falsifying data, or what the goal would be. as the deck tracker already has the ability to import decks from the web (currently hearthpwn)

from the point of memory allocation (constructors, destructors, and assignments) taking one image of the decklist as it exists holding that in memory as an object, and then for the prospectCard have a seperate memory object and just keep reassigning that would be the most efficient in my mind.

**Comment by culinko** [comment]
Importing should be fixed in 0.15 so this should no longer be an issue.
